src/overlays/actors/ovl_Boss_Goma/z_boss_goma.c

```diff
@@ -21,115 +111,1836 @@ const ActorInit Boss_Goma_InitVars = {
+}
+
+void BossGoma_SetupWallClimb(BossGoma* this) {
+    SkelAnime_ChangeAnim(&this->skelanime, &D_060066DC, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_060066DC.genericHeader),
+                         0, -10.0f);
+    this->actionFunc = BossGoma_WallClimb;
+    this->actor.speedXZ = 0.0f;
+    this->actor.velocity.y = 0.0f;
+    this->actor.gravity = 0.0f;
+}
+
+/**
+ * Gohma either reached the ceiling after climbing a wall, or is waiting for the player to kill the (children) Gohmas.
+ */
+void BossGoma_SetupCeilingMoveToCenter(BossGoma* this) {
+    SkelAnime_ChangeAnim(&this->skelanime, &D_06002360, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_06002360.genericHeader),
+                         0, -5.0f);
+    this->actionFunc = BossGoma_CeilingMoveToCenter;
+    this->actor.speedXZ = 0.0f;
+    this->actor.velocity.y = 0.0f;
+    this->actor.gravity = 0.0f;
+    this->framesUntilNextAction = Math_Rand_S16Offset(30, 60);
+}
+
+/**
+ * Gohma approaches the player as long as it has patience (see patienceTimer), then moves away from the player
+ * Gohma climbs any wall it collides with
+ */
+void BossGoma_SetupFloorMain(BossGoma* this) {
+    SkelAnime_ChangeAnim(&this->skelanime, &D_0601EB4C, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_0601EB4C.genericHeader),
+                         0, -5.0f);
+    this->actionFunc = BossGoma_FloorMain;
+    this->framesUntilNextAction = Math_Rand_S16Offset(70, 110);
+}
+
+/**
+ * Gohma is back on the floor after the player has killed its children Gohmas.
+ */
+void BossGoma_SetupFloorLand(BossGoma* this) {
+    SkelAnime_ChangeAnim(&this->skelanime, &D_06005F28, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_06005F28.genericHeader),
+                         2, -2.0f);
+    this->actionFunc = BossGoma_FloorLand;
+    this->currentAnimFrameCount = SkelAnime_GetFrameCount(&D_06005F28.genericHeader);
+}
+
+/**
+ * Gohma is back on the floor after the player struck it down from the ceiling.
+ * Gohma ends up stunned (BossGoma_FloorStunned)
+ */
+void BossGoma_SetupFloorLandStruckDown(BossGoma* this) {
+    SkelAnime_ChangeAnim(&this->skelanime, &D_060058C0, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_060058C0.genericHeader),
+                         2, -2.0f);
+    this->currentAnimFrameCount = SkelAnime_GetFrameCount(&D_060058C0.genericHeader);
+    this->actionFunc = BossGoma_FloorLandStruckDown;
+    this->currentAnimFrameCount = SkelAnime_GetFrameCount(&D_060058C0.genericHeader);
+}
+
+/**
+ * Gohma is stunned and vulnerable. It can only be damaged during this action.
+ */
+void BossGoma_SetupFloorStunned(BossGoma* this) {
+    SkelAnime_ChangeAnim(&this->skelanime, &D_0600CCD8, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_0600CCD8.genericHeader),
+                         0, -2.0f);
+    this->actionFunc = BossGoma_FloorStunned;
+}
+
+/**
+ * Take an attack posture when the player is close enough.
+ * If the player backs off, cancel the attack, or attack.
+ */
+void BossGoma_SetupFloorAttackPosture(BossGoma* this) {
```

Comment:
this sounds like more of a description of the action itself no? not the setup

Reply:
same goes for all of the comments on these setup funcs i guess

Reply:
needs discussion, I agree some of these comments would fit better on actual actions rather than setups but others only work on setups and I dont feel like skipping comments on some setups to put them on the action but idk really